doc/rtd/explanation/about-cloud-config.rst

```diff
@@ -0,0 +1,163 @@
+.. _about-cloud-config:
+
+About the cloud-config file
+***************************
+
+The ``#cloud-config`` file is a type of user data that cloud-init can consume
+to automatically set up various aspects of the system. It is the preferred way
+to pass this type of configuration to cloud-init, and is commonly referred to
+as "cloud config".
+
+Note that networks are not configurable via the ``#cloud-config`` file because
+:ref:`network configuration <network_config>` comes from the cloud.
+
+How do I create a cloud-config file?
+====================================
+
+The cloud-config file uses `YAML version 1.1`_. The file is composed of a
+**header** and one or more **modules**.
+
+* **The header**:
+  The first line **must start** with ``#cloud-config``. This line identifies
+  the file to cloud-init and ensures that it will be processed as intended.
+
+* **The modules**:
+  After the header, every aspect of the system's configuration is controlled
+  through specific cloud-init modules.
+
+Each module is declared through the use of its **top-level key**, and the
```

Comment:
It would be nice if we were this consistent, but not all modules have a single top-level key. E.g., the [set passwords](https://cloudinit.readthedocs.io/en/latest/reference/modules.html#set-passwords) module reacts to the `password`, `chpasswd`, and `ssh_pwauth` keys.

Additionally, there are multiple modules that may react to the same key. E.g., `write_files` key is used by both the `write_files` and `write_files_deferred` module and `hostname` and `fqdn` keys are used by both the `set_hostname` and `update_hostname` modules.